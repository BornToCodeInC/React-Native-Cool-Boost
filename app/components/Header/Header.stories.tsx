import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { object, text, color, boolean } from '@storybook/addon-knobs';
import { Header } from './Header';

// export default {
//     title: 'components/Header',
//     component: Header,
// };
//
// export const Basic = () => (
//     <Header />
// );

// Basic.args = {};

storiesOf('Header', module)
    .addDecorator(story => <View style={s.decorator}>{story()}</View>)
    // 👇 you can add multiple variants of component, here's variant with name 'default'
    .add('default', () => (
        <Header>
            <Text>😀 😎 👍 💯</Text>
        </Header>
    ));

const s = StyleSheet.create({
    decorator: {
        flex: 1,
        justifyContent: 'center',
        padding: 16,
    },
});