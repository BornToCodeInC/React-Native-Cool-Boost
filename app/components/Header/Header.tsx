import React from 'react';
import {StyleSheet, View} from 'react-native';

export const Header: React.FC<React.ReactNode> = ({children}): JSX.Element => {
  return <View style={styles.container}>{children}</View>;
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 20,
    backgroundColor: '#008ACE',
    width: '100%',
    minHeight: 55,
    zIndex: 1,
  },
});
