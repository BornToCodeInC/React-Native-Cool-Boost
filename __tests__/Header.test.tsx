import 'react-native';
import React from 'react';
import { Header } from '../app/components/Header/Header';
import { render, waitFor } from '@testing-library/react-native';
import { Text } from 'react-native';

describe('Header.test.tsx', () => {
    test('renders correctly', () => {
        const component = render(<Header/>);
        const snapshot = component.toJSON();
        expect(snapshot).toMatchSnapshot();
    });

    test('should render properly content', async () => {
        const { getByText } = render(<Header><Text>Internal</Text></Header>);
        await waitFor(() => expect(getByText('Internal')).toBeTruthy());
    });
})

