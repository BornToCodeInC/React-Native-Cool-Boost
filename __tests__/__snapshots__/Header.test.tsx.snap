// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.test.tsx renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#008ACE",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 55,
      "paddingHorizontal": 20,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 1,
    }
  }
/>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#008ACE",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 55,
      "paddingHorizontal": 20,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 1,
    }
  }
/>
`;
